### ISSUE TITLE
Stopped working for me

### ISSUE THREAD

**Issue by dsandstrom** [hmatsuda/ruby-block#16]
Atom v0.202 - OSX.  I think it stopped working on v0.200.  I also notice a slow down when moving to an `end` in RSpec.  Like when I move down from the second line to the third: 

``` rspec
it "does something" do
  expect(true).to eq(true)
end
```

The delay is gone if I disable this package.

**Comment by hmatsuda** [comment]
When Atom v0.209 on OSX,  I can't reproduce this issue..

**Comment by dsandstrom** [comment]
Works for me now, too.

**Comment by dsandstrom** [comment]
I spoke too soon, stopped working.  But, worked again after closing Atom and reopening.  I will try to figure out the repro steps.

**Comment by dsandstrom** [comment]
However, the delay when going to the end line in RSpec code seems to be gone.

**Comment by dsandstrom** [comment]
It has something to do with describe and context blocks.  They don't seem to be supported and are throwing off the do-end count.  I will try to make an example file that you can use to reproduce.

**Comment by ypresto** [comment]
Same problem. I'm using Atom 1.0.2.

**Comment by hmatsuda** [comment]
@ypresto I can't reproduce this issue in my environment so I also use 1.0.2 on OSX. I would be helpful if you describe following informations:
- platform
- reproduce code
- installed packages
- using ui and sytax theme
- config.cson
- styles.less

Thanks.

**Comment by ypresto** [comment]
It randomly occurs when moving cursor fast around blocks.
Switch to another tab does not fix problem.
Restarting fixes problem.
- platform: OS X 10.10.4
- reproduce code: Not ensured. Moving cursor in rspec something like below.

``` ruby
describe '#call' do
  subject { described_class.new(user, family).call(to_foo, foo_invitation) }

  context 'call with invalid argument' do
    it "raise argument error" do
      expect { described_class.new(user, family).call(to_foo, 'foobar') }.to raise_error ArgumentError
    end
  end
end
```
- installed packages (ls ~/atom/packages)

```
README.md          auto-indent           css-snippets      git-blame-plus    hashrocket           language-gradle  linter-coffeelint  linter-rubocop    multi-cursor  regex-railroad-diagram  ruby-block-converter  vim-mode
advanced-new-file  auto-update-packages  django-templates  git-diff-details  highlight-selected   language-groovy  linter-haml        linter-ruby       open-recent   robe                    ruby-define-method    vim-surround
atom-beautify      autocomplete-paths    emmet             git-log           htmlhint             language-haml    linter-jscs        linter-scss-lint  pretty-json   rspec                   ruby-slim
atom-django        autocomplete-python   file-icons        git-plus          japanese-wrap        language-rspec   linter-jshint      linter-slim       python-isort  rspec-focus             scope-inspector
atom-ternjs        color-picker          gist-it           gulp-snippets     javascript-snippets  linter           linter-pylint      merge-conflicts   python-jedi   ruby-block              sync-settings
```
- ui theme: Atom Dark
- syntax theme: One Dark
- config.cson

``` cson
"*":
  "exception-reporting":
    userId: "..."
  welcome:
    showOnStartup: false
  core:
    projectHome: "/Users/yuya/repo/github.com"
    disabledPackages: [
      "fancy-new-file"
      "minimap-highlight-selected"
      "minimap"
      "autocomplete-plus-async"
    ]
    themes: [
      "atom-dark-ui"
      "one-dark-syntax"
    ]
    excludeVcsIgnoredPaths: false
  editor:
    invisibles: {}
    zoomFontWhenCtrlScrolling: true
    showInvisibles: true
    showIndentGuide: true
    softWrap: true
  "sync-settings":
    personalAccessToken: "..."
    gistId: "..."
  "scope-inspector":
    userId: "..."
    showSidebar: false
  minimap:
    plugins:
      "highlight-selected": true
  "recent-files":
    updated: false
  "autocomplete-plus":
    navigateCompletions: "ctrl-p,ctrl-n"
    confirmCompletion: "tab"
  "atom-beautify":
    _analyticsUserId: "..."
    js_brace_style: "expand"
  "atom-ternjs":
    urls: false
    coffeeScript: true
    inlineFnCompletion: false
  "autocomplete-plus-async":
    autoActivationDelay: 1000
  "vim-surround":
    surroundCommand: "S"
  "linter-jscs":
    onlyConfig: true
  "vim-mode":
    useSmartcaseForSearch: true
  "release-notes":
    viewedVersion: "0.71.0"
  metrics:
    userId: "..."
  rspec:
    command: "bin/rspec"
  "file-icons":
    coloured: false
  "gist-it":
    newGistsDefaultToPrivate: true
    userToken: "..."
  autosave:
    enabled: true
  "rails-transporter":
    viewFileExtension: "html.slim"
  "atom-ctags":
    cmdArgs: "--languages=ruby,coffee --exclude=.git"
    buildTimeout: 30000
    autoBuildTagsWhenActive: true
  "python-jedi":
    Pathtopython: "/Users/yuya/.anyenv/envs/pyenv/versions/3.4.3/bin/python"
    enablePathtopython: true
  hashrocket:
    pythonExecutablePath: "/Users/yuya/.anyenv/envs/pyenv/shims/python"
  "tree-view":
    hideVcsIgnoredFiles: true
  tabs:
    usePreviewTabs: true
  "linter-slim":
    rails: true
    executableDir: "/Users/yuya.tanaka/.anyenv/envs/rbenv/versions/2.2.2/bin"
  linter:
    showErrorTabLine: true
  "linter-ruby": {}
  "linter-rubocop": {}
  "autocomplete-python": {}
```
- styles.less: Default

**Comment by hmatsuda** [comment]
@ypresto Thanks for describing your atom setting information! I emulate your environment but it doesn't occur in my computer. If you find reproduce step, please let me know.

**Comment by dsandstrom** [comment]
Reproduce steps
1) make a new file named **test_spec.rb**
2) Copy @ypresto's code above into file
3) Set syntax to RSpec
4) Put cursor on the end of line 7 (`it`'s `end`)
5) Move cursor down

Notice `context` line is not highlighted.  After moving up, `it`'s line is not highlighted.

I noticed the bug only happens for the RSpec syntax.  If the syntax is Ruby, no problem.

**Comment by hmatsuda** [comment]
Thanks @ypresto and @dsandstrom. I fixed this issue and published v0.3.5 with the bugfixes. If this will rearise, please reopen the issue.
